src/node/setupRemoteServer.ts

```diff
@@ -0,0 +1,402 @@
+    super(...handlers)
+
+    this.handlersController = new AsyncHandlersController({
+      storage: remoteHandlersContext,
+      initialHandlers: handlers,
+    })
+
+    this.executionContexts = new Map()
+  }
+
+  get serverUrl(): URL {
+    invariant(
+      this[kServerUrl],
+      'Failed to get a remote port in `setupRemoteServer`. Did you forget to `await remote.listen()`?',
+    )
+
+    return this[kServerUrl]
+  }
+
+  get boundaryId(): string {
+    const context = remoteHandlersContext.getStore()
+
+    invariant(
+      context != null,
+      'Failed to get "contextId" on "SetupRemoteServerApi": no context found. Did you forget to wrap this closure in `remote.boundary()`?',
+    )
+
+    return context.boundaryId
+  }
+
+  public async listen(options: Partial<ListenOptions> = {}): Promise<void> {
+    this.resolvedOptions = mergeRight(
+      DEFAULT_LISTEN_OPTIONS,
+      options,
+    ) as RequiredDeep<ListenOptions>
+    const dummyEmitter = new Emitter<LifeCycleEventsMap>()
+
+    const server = await createSyncServer()
+    this[kServerUrl] = getServerUrl(server)
+
+    process
+      .once('SIGTERM', () => closeSyncServer(server))
+      .once('SIGINT', () => closeSyncServer(server))
+
+    // Close the server if the setup API is disposed.
+    this.subscriptions.push(() => closeSyncServer(server))
+
+    server.on('request', async (incoming, outgoing) => {
+      if (!incoming.method) {
+        return
+      }
+
+      // Handle the handshake request from the client.
+      if (incoming.method === 'HEAD') {
+        outgoing.writeHead(200).end()
+        return
+      }
+
+      // Handle life-cycle event requests forwarded from `setupServer`.
+      if (incoming.url === '/life-cycle-events') {
+        this.handleLifeCycleEventRequest(incoming, outgoing)
+        return
+      }
+
+      const requestId = incoming.headers['x-msw-request-id']
+      const requestUrl = incoming.headers['x-msw-request-url']
+      const contextId = incoming.headers['x-msw-boundary-id']
+
+      if (typeof requestId !== 'string') {
+        outgoing.writeHead(400)
+        outgoing.end('Expected the "x-msw-request-id" header to be a string')
+        return
+      }
+
+      if (typeof requestUrl !== 'string') {
+        outgoing.writeHead(400)
+        outgoing.end('Expected the "x-msw-request-url" header to be a string')
+        return
+      }
+
+      // Validate remote context id.
+      if (contextId != null && typeof contextId !== 'string') {
+        outgoing.writeHead(400)
+        outgoing.end(
+          `Expected the "contextId" value to be a string but got ${typeof contextId}`,
+        )
+        return
+      }
+
+      const request = new Request(requestUrl, {
+        method: incoming.method,
+        body:
+          incoming.method !== 'HEAD' && incoming.method !== 'GET'
+            ? (Readable.toWeb(incoming) as ReadableStream<unknown>)
+            : null,
+      })
```

Comment:
```suggestion
        duplex: incoming.method !== 'HEAD' && incoming.method !== 'GET'
            ? 'half'
            : undefined,
      })
```
We can't send a stream in the body without explicitly defining the duplex, otherwise you get `Uncaught TypeError: RequestInit: duplex option is required when sending a body.`. See https://github.com/nodejs/node/issues/46221

Reply:
Oh, it's this issue again! Thanks for pointing it out.
We can also set `duplex` all the time, as it doesn't seem to error in case of requests without any body.

Reply:
Pushed the fix at 3c76156efe769f6d9406314909e82070f90d8649

Reply:
I'm also getting the following error, followed by a server crash, possibly because the request body is being used elsewhere:

```
TypeError: unusable
    at Request.clone
    at serializeFetchRequest
    at handleLifeCycleEvent
```

Reply:
Looks like the ReadableStream is closed: `ReadableStream { locked: false, state: 'closed', supportsBYOB: false }`

Reply:
Can you tell me more about the use case? I'm not hitting this one.